util/codec/codec.go

```diff
@@ -352,6 +353,232 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 	return
 }
 
+// HashChunkColumns writes the encoded value of each row's column, which of index `colIdx`, to h.
+func HashChunkColumns(sc *stmtctx.StatementContext, h []hash.Hash64, chk *chunk.Chunk, tp *types.FieldType, colIdx int, buf []byte, isNull []bool) (err error) {
+	const (
+		sizeUint64  = unsafe.Sizeof(uint64(0))
+		sizeFloat64 = unsafe.Sizeof(float64(0))
+	)
+	var (
+		b   []byte
+		row chunk.Row
+	)
+	column := chk.Column(colIdx)
+	rows := chk.NumRows()
+	switch tp.Tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeYear:
+		for i := 0; i < rows; i++ {
+			row = chk.GetRow(i)
+			if row.IsNull(colIdx) {
+				buf[0], b = NilFlag, nil
+				isNull[i] = true
+			} else {
+				buf[0] = varintFlag
+				if mysql.HasUnsignedFlag(tp.Flag) {
+					if integer := row.GetInt64(colIdx); integer < 0 {
+						buf[0] = uvarintFlag
+					}
+				}
+				b = row.GetRaw(colIdx)
```

Comment:
We can replace each `row.GetRaw` to [`column.GetRaw`](https://github.com/pingcap/tidb/blob/1b0f61a/util/chunk/column.go#L541) in this function, so `row = chk.GetRow(i)` can be left out.